### ISSUE TITLE
Add DNS over HTTP/2 Stub Resolver

### ISSUE THREAD

**Issue by dknecht** [cloudflare/cloudflared#4]
Create an option for argo tunnel to start up as DNS stub resolver.

Design
- Locally-running application that listens on localhost on a specific port and accepts incoming DNS requests
- Queries are sent to Cloudflare's resolver via the DoH protocol and responses to the application as DNS messages

Required
- Option to run independently and in parallel to existing warp functionality.
- Option to set an IP (IPv4/IPv6) and a port.
- Expose metrics via existing prometheus metrics exporter
- Defaults to https://dns.cloudflare.com/.well-known/dns-query

 
Nice to have
- Caching
- Similar fallback option to Firefox.

1. race OS vs TRR in parallel and take the faster one
2. TRR first.. generally use TRR and only go to the native resolver for resolution of trr fails.
3. TRR only.. like first, but never fallback
4. Shadow mode.. like race but always use OS result. (this is how we will do the initial tests)

If you are using mode 3 you have a bootstrap problem connecting to the DoH endpoint (in the other modes the bootstrap falls back to native dns) but you can use the network.trr.bootstrapAddress configuration field to "pre-resolve" the IP address of the DoH server (or you could use an IP based cert I suppose).
